File: app/src/main/java/com/esorokin/lantector/ui/plugins/ProgressPlugin.java
package com.esorokin.lantector.ui.plugins;

import android.app.ProgressDialog;

import com.esorokin.lantector.R;
import com.esorokin.lantector.ui.plugins.base.BaseDependencyPlugin;

public class ProgressPlugin extends BaseDependencyPlugin<ContextProvider> {
	private ProgressDialog dialog;

	public ProgressPlugin(ContextProvider delegate) {
		super(delegate);
	}

	@Override
	public void onDestroy() {
		super.onDestroy();
		hideProgress();
	}

	public void showProgress() {
		hideProgress();
		dialog = new ProgressDialog(getDependency().getContext());
		dialog.setIndeterminate(true);
		dialog.setCancelable(false);
		dialog.setCanceledOnTouchOutside(false);
		dialog.setMessage(getDependency().getContext().getString(R.string.loading));
		dialog.show();
	}

	public void hideProgress() {
		if (dialog != null) {
			dialog.dismiss();
		}
	}
}

File: app/src/main/java/com/esorokin/lantector/ui/plugin/ProgressPlugin.kt
package com.esorokin.lantector.ui.plugin

import android.app.ProgressDialog
import android.content.Context
import com.esorokin.lantector.R
import com.esorokin.lantector.ui.plugin.base.BasePlugin

class ProgressPlugin(private val contextProvider: () -> Context) : BasePlugin() {
    private var dialog: ProgressDialog? = null

    override fun onDestroy() {
        super.onDestroy()
        hideProgress()
    }

    fun showProgress() {
        hideProgress()
        dialog = ProgressDialog(contextProvider.invoke()).apply {
            isIndeterminate = true
            setCancelable(true)
            setCanceledOnTouchOutside(false)
            setMessage(context.getString(R.string.loading))
            show()
        }
    }

    fun hideProgress() {
        dialog?.dismiss()
    }
}
